src/App.jsx

```diff
@@ -15,29 +15,49 @@ export default class App extends React.Component {
     return firebase.database().ref(path);
   }
 
+  static budgetRef() {
+    return App.getRef(`/users/${TEST_USER_ID}/budgets/${TEST_BUDGET_ID}`);
+  }
+
   constructor(props) {
     super(props);
     this.state = {
-      transactions: props.transactions,
+      transactions: {},
     };
 
     this.addTransaction = this.addTransaction.bind(this);
     this.editTransaction = this.editTransaction.bind(this);
   }
 
+  componentDidMount() {
+    return App.budgetRef().child('transactions').once('value')
```

Comment:
Eventually, we should limit this to all transactions within the last 30 days or something like that.